kaiax/gasless/impl/tx_pool.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2025 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package impl
+
+import (
+	"math"
+
+	"github.com/kaiachain/kaia/blockchain/types"
+	"github.com/kaiachain/kaia/kaiax"
+)
+
+var _ kaiax.TxPoolModule = (*GaslessModule)(nil)
+
+func (g *GaslessModule) PreAddLocal(tx *types.Transaction) error {
+	return nil
+}
+
+func (g *GaslessModule) PreAddRemote(tx *types.Transaction) error {
+	return nil
+}
+
+func (g *GaslessModule) IsModuleTx(tx *types.Transaction) bool {
+	return g.IsApproveTx(tx) || g.IsSwapTx(tx)
+}
+
+func (g *GaslessModule) GetCheckBalance() func(tx *types.Transaction) error {
+	return func(*types.Transaction) error { return nil }
+}
+
+func (g *GaslessModule) IsReady(txs map[uint64]*types.Transaction, i uint64, ready types.Transactions) bool {
+	tx := txs[i]
+	addr := tx.ValidatedSender()
+	nonce := g.StateDB.GetNonce(addr)
+
+	if i > nonce+1 {
+		return false
+	}
+
+	isApproveTx := g.IsApproveTx(tx)
+	isSwapTx := g.IsSwapTx(tx)
+
+	if isApproveTx && i == nonce && i+1 < uint64(math.MaxUint64) {
+		if next := txs[i+1]; next != nil && g.IsSwapTx(next) {
+			return g.IsExecutable(tx, next)
+		}
+	}
```

Comment:
How about introducing a new function for this branch?
```
// IsApproveTxReady assumes that the caller checked `g.IsApproveTx(approveTx)`
func (g *GaslessModule) IsApproveTxReady(approveTx, swapTx *types.Transaction) bool {
	nonce := approveTx.nonce
	if nonce != statedb.nonce {
		return false
	}
	if nonce == uint64(math.MaxUint64) {
		return false
	}
	if swapTx == nil || g.IsSwapTx(swapTx) {
		return false
	}
	
	return g.IsExecutable(approveTx, swapTx)
}
```

Reply:
Do you assume this is used in tx pool?

Reply:
No, only from `IsReady`

Reply:
fix at https://github.com/kaiachain/kaia/pull/263/commits/a15e7868058165b585799dc051d5086cdf6a40b5